### PR TITLE
Update to Cluster API v1.2.3 in go dependencies

### PR DESCRIPTION
**Pull request by randomvariable**
### What this PR does / why we need it
- Update Cluster API and related dependencies to v1.2.3

The providers itself will be updated in a separate PR (#3513). The API changes are compatible.

Fixes # N/A

### Release note

```release-note
Update Cluster API go dependencies to v1.2.3, controller-runtime to 0.12 and Kubernetes to 1.24
For AWS, the following actions have been added to the policy for Cluster API AWS controllers:
* ec2:ReplaceRoute
* elasticloadbalancing:DeleteTargetGroup
```

### CODE REVIEW

File: pkg/v1/tkg/client/machine_deployment_cc.go [hunk 1]
```diff
@@ -332,7 +332,7 @@ func DoDeleteMachineDeploymentCC(clusterClient clusterclient.Client, cluster *ca
 
 	cluster.Spec.Topology.Workers.MachineDeployments = mds
 
-	if err := clusterClient.UpdateResource(cluster, cluster.ClusterName, cluster.Namespace); err != nil {
+	if err := clusterClient.UpdateResource(cluster, cluster.Name, cluster.Namespace); err != nil {
```
Comment (randomvariable) [inline pkg/v1/tkg/client/machine_deployment_cc.go:335]: is this correct @sbueringer  @ykakarap ?

**Comment by randomvariable** [comment]
E2Es are likely to fail because Tanzu's kind image is out of date.

File: pkg/v1/tkg/client/machine_deployment_cc.go (hunk 1 above)
Comment (sbueringer) [reply pkg/v1/tkg/client/machine_deployment_cc.go:335]: Assuming that you want to hand over the name of the cluster - yes.

.ClusterName was always an empty string

**Comment by sbueringer** [comment]
Lgtm so far

**Comment by randomvariable** [comment]
makefile fixes are going in https://github.com/vmware-tanzu/tanzu-framework/pull/3159

File: cli/runtime/hack/tools/go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module github.com/vmware-tanzu/hack/tools
+module github.com/vmware-tanzu/cli/runtime/hack/tools
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply cli/runtime/hack/tools/go.mod:1]: https://github.com/vmware-tanzu/tanzu-framework/pull/3285 .

would appreciate it being merged.

File: tkg/kind/client_test.go [hunk 3]
```diff
@@ -19,6 +19,9 @@ import (
 	"github.com/vmware-tanzu/tanzu-framework/tkg/constants"
 	"github.com/vmware-tanzu/tanzu-framework/tkg/kind"
 	"github.com/vmware-tanzu/tanzu-framework/tkg/utils"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
Comment (sbueringer) [inline tkg/kind/client_test.go:24]: Was this intended? (given that there is no other diff in this file)

Comment (randomvariable) [reply tkg/kind/client_test.go:24]: Yup. For some reason it was missing and this file was otherwise broken.

Comment (sbueringer) [reply tkg/kind/client_test.go:24]: Interesting that that didn't surface in some way

Comment (randomvariable) [reply tkg/kind/client_test.go:24]: Yeah, i need to take a look at why.

**Comment by randomvariable** [comment]
looks like i've broken pinniped somewhere, will take a look

**Comment by randomvariable** [comment]
Attempting to rebase on main breaks Tanzu Framework as packages have not been completely relocated.

**Comment by randomvariable** [comment]
I can't run CAPD successfully at all locally, so not sure where to go from here.

**Comment by randomvariable** [comment]
Tests passed locally, so not sure what's up.

**Review by anujc25 — COMMENTED** [review]
LGTM.

**Comment by alfredthenarwhal** [comment]
Tests failed! Build no: 2842

**Comment by alfredthenarwhal** [comment]
Tests failed! Build no: 2843

**Comment by vuil** [comment]
Thanks for the changes. Looks like the two unit tests failures could be related to the aws sdk version change
https://github.com/vmware-tanzu/tanzu-framework/actions/runs/3200808472/jobs/5228136568#step:14:2703

**Comment by randomvariable** [comment]
Ah, we have new permissions from CAPA upstream. will need to update fixtures and Changelog

**Comment by randomvariable** [comment]
Changes made, and release note updated.

**Comment by vuil** [comment]
LGTM, but trying to understand what happend with the Main/Build check.

**Comment by vuil** [comment]
Looks like some kind of runner OOS issue, retriggering now.

**Review by imikushin — APPROVED** [review]
LGTM!

**Review by vuil — APPROVED** [review]

**Pull request merged**